service/dap/server.go

```diff
@@ -442,9 +455,135 @@ func (s *Server) onLaunchRequest(request *dap.LaunchRequest) {
+				FailedToLaunch, "Failed to launch",
+				fmt.Sprintf("Unsupported 'backend' value %q in debug configuration.", backend))
+			return
+		}
+
+		// Record only works on rr trace directories
+		traceDirectory, ok := request.Arguments["traceDirectory"].(string)
+		if !ok || traceDirectory == "" {
+			s.sendErrorResponse(request.Request,
+				FailedToLaunch, "Failed to launch",
+				"The traceDirectory attribute is missing in debug configuration.")
+			return
+		}
+		if _, err := os.Stat(traceDirectory); os.IsNotExist(err) {
+			s.sendErrorResponse(request.Request,
+				FailedToLaunch, "Failed to launch",
+				"The traceDirectory attribute points to an directory that does not exist.")
+			return
+		}
+
+		buildFlags := ""
+		buildFlagsArg, ok := request.Arguments["buildFlags"]
+		if ok {
+			buildFlags, ok = buildFlagsArg.(string)
+			if !ok {
+				s.sendErrorResponse(request.Request,
+					FailedToLaunch, "Failed to launch",
+					fmt.Sprintf("'buildFlags' attribute '%v' in debug configuration is not a string.", buildFlagsArg))
+				return
+			}
+		}
+
+		output, ok := request.Arguments["output"].(string)
+		if !ok || output == "" {
+			output = debugBinary
+		}
+		debugname, err := filepath.Abs(output)
+		err = gobuild.GoBuild(debugname, []string{program}, buildFlags)
+
+		if err != nil {
+			s.sendErrorResponse(request.Request,
+				FailedToLaunch, "Failed to build recording target",
+				fmt.Sprintf("Build error: %s", err.Error()))
+			return
+		}
+	}
+
+	if mode == "coredump" {
+		backend, ok := request.Arguments["backend"].(string)
+		if !ok || backend == "" {
+			s.sendErrorResponse(request.Request,
+				FailedToLaunch, "Failed to launch",
+				"The backend attribute is missing debug configuration.")
+			return
+		}
+
+		// Validate backend for mode
+		if backend != "coredump" {
+			s.sendErrorResponse(request.Request,
+				FailedToLaunch, "Failed to launch",
+				fmt.Sprintf("Unsupported 'backend' value %q in debug configuration.", backend))
+			return
+		}
+
+		// Coredump only works on core dump files
+		coreDumpFile, ok := request.Arguments["coreDumpPath"].(string)
+		if !ok || coreDumpFile == "" {
+			s.sendErrorResponse(request.Request,
+				FailedToLaunch, "Failed to launch",
+				"The coreDumpPath attribute is missing in debug configuration.")
+			return
+		}
+		if _, err := os.Stat(coreDumpFile); os.IsNotExist(err) {
+			s.sendErrorResponse(request.Request,
+				FailedToLaunch, "Failed to launch",
+				"The coreDumpPath attribute points to an directory that does not exist.")
+			return
+		}
+
+		s.config.Debugger.CoreFile = coreDumpFile
+		s.config.Debugger.Backend = backend
+	}
+
+	if mode == "replay" {
+		backend, ok := request.Arguments["backend"].(string)
+		if !ok || backend == "" {
+			s.sendErrorResponse(request.Request,
+				FailedToLaunch, "Failed to launch",
+				"The backend attribute is missing debug configuration.")
+			return
+		}
+
+		// Replay only works on rr trace directories
+		if backend != "rr" {
```

Comment:
I corrected the code, the backend depends on the command (`dlv replay` replays traces and `dlv core` replays core dumps). With my last push I corrected the code so that we work with a single replay mode and the `rr` and `core` backends, which are analog to these commands (related to this response https://github.com/go-delve/delve/pull/2367#discussion_r591998098)